cloud/satellite.go

```diff
@@ -198,37 +198,60 @@ func (c *Client) ReserveSatellite(ctx context.Context, name, orgName, gitAuthor,
 		ctxTimeout, cancel := context.WithTimeout(ctx, 10*time.Minute)
 		defer cancel()
 		defer close(out)
-		stream, err := c.compute.ReserveSatellite(c.withAuth(ctxTimeout), &pb.ReserveSatelliteRequest{
-			OrgId:          orgID,
-			Name:           name,
-			CommitEmail:    gitAuthor,
-			GitConfigEmail: gitConfigEmail,
-			IsCi:           isCI,
-		})
-		if err != nil {
-			out <- SatelliteStatusUpdate{Err: errors.Wrap(err, "failed opening satellite reserve stream")}
-			return
-		}
-		for {
-			update, err := stream.Recv()
-			if err == io.EOF {
-				return
-			}
+		for i := 1; i <= 5; i++ {
+			stream, err := c.compute.ReserveSatellite(c.withRetryCount(c.withAuth(ctxTimeout), i), &pb.ReserveSatelliteRequest{
+				OrgId:          orgID,
+				Name:           name,
+				CommitEmail:    gitAuthor,
+				GitConfigEmail: gitConfigEmail,
+				IsCi:           isCI,
+			})
 			if err != nil {
-				out <- SatelliteStatusUpdate{Err: errors.Wrap(err, "failed receiving satellite reserve update")}
+				out <- SatelliteStatusUpdate{Err: errors.Wrap(err, "failed opening satellite reserve stream")}
 				return
 			}
-			status := satelliteStatus(update.Status)
-			if status == SatelliteStatusFailed {
-				out <- SatelliteStatusUpdate{Err: errors.New("satellite is in a failed state")}
-				return
+			var lastStatus string
+			for {
+				update, err := stream.Recv()
+				if err == io.EOF {
+					if !isFinalStatus(lastStatus) {
+						// Go got EOF, but the status doesn't seem appropriate => retry
+						time.Sleep(time.Duration(i) * time.Second)
+						break
+					}
+					return
+				}
+				if err != nil {
+					out <- SatelliteStatusUpdate{Err: errors.Wrap(err, "failed receiving satellite reserve update")}
+					return
+				}
+				lastStatus = satelliteStatus(update.Status)
+				if lastStatus == SatelliteStatusFailed {
+					out <- SatelliteStatusUpdate{Err: errors.New("satellite is in a failed state")}
+					return
+				}
+				out <- SatelliteStatusUpdate{State: satelliteStatus(update.Status)}
 			}
-			out <- SatelliteStatusUpdate{State: satelliteStatus(update.Status)}
 		}
+		// max retries consumed
+		out <- SatelliteStatusUpdate{Err: errors.Wrap(err, "failed to receive a valid final satellite status")}
+		return
+
 	}()
 	return out
 }
 
+func isFinalStatus(status string) bool {
+	switch status {
+	case SatelliteStatusOperational:
+		return true
+	case SatelliteStatusFailed:
+		return true
```

Comment:
```suggestion
	case SatelliteStatusOperational, case SatelliteStatusFailed:
		return true
```

Nit: cases can be combined.